modules/rostests/rosautotest/main.cpp

```diff
@@ -52,6 +53,9 @@ wmain(int argc, wchar_t* argv[])
 {
     int ReturnValue = 1;
 
+    wcscpy(TestName, argv[0]);
+    wcsncat (TestName, L".exe", 4);
```

Comment:
These calls are unsafe.

Reply:
Also you should use `GetModuleFileNameW(NULL, TestName, _countof(TestName));`

Reply:
I like the GetModuleFineNameW idea and changed the code to reflect this method. Please have a look. Also, I used a string safe function to copy the truncated file name to TestName.